explainaboard/metrics/metric.py

```diff
@@ -45,7 +46,7 @@ def to_dict(self):
 
 
 @dataclass
-class MetricConfig(dict):
+class MetricConfig(SerializableDataclass):
```

Comment:
@odashi (@neubig) I'll try to remove `external_stats` in a separate PR first, and then back to this PR since the required change would include irrelevant changes.